### PR TITLE
features: save and get username

### PR DESCRIPTION
**Pull request by PanVova**

### CODE REVIEW

**Comment by PanVova** [comment]
I had also idea about not using usecase component and provide datastore component right to viewModel. I am not sure about this idea so it's better to write a comment under this message.

File: menuFeature/src/main/java/com/panhuk/menufeature/MenuFragment.kt [hunk 1]
```diff
@@ -90,22 +90,31 @@ class MenuFragment : Fragment() {
...
+      Text(stringResource(textId), fontSize = 24.sp)
+    }
+  }
+
+  private fun navigateToFragment(textId: Int) {
+    val navigator = (requireActivity() as MenuNavigator)
+    when (textId) {
+      R.string.play -> navigator.navigateMenuToPlayFragment()
+      R.string.settings -> navigator.navigateMenuToSettingsFragment()
+      R.string.leaderboard -> navigator.navigateMenuToLeaderboardFragment()
```
Comment (NickHuk) [inline menuFeature/src/main/java/com/panhuk/menufeature/MenuFragment.kt]: handle click based on resource id is bad idea, ass lambda instead or use enum

File: playFeature/src/main/java/com/panhuk/playfeature/di/PlayComponent.kt [hunk 2]
```diff
@@ -30,7 +30,7 @@ interface PlayComponent {
     fun create(context: Context): PlayComponent =
       DaggerPlayComponent.builder()
         .sessionTokenRepoComponent(SessionTokenRepoComponent.create(context))
-        .useCaseComponent(UseCaseComponent.create())
+        .useCaseComponent(UseCaseComponent.create(context))
```
Comment (NickHuk) [inline playFeature/src/main/java/com/panhuk/playfeature/di/PlayComponent.kt]: context in use case WTF?

File: usecaseImpl/build.gradle [hunk 3]
```diff
@@ -39,4 +39,5 @@ dependencies {
   implementation project(':domain')
   implementation project(':repository')
   implementation project(':useCase')
+  implementation project(':dataSource')
```
Comment (NickHuk) [inline usecaseImpl/build.gradle]: why dataSource is here i think it's redundant

File: settingsFeature/src/main/java/com/panhuk/settingsfeature/SettingsViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    usernameUseCase.getUsername().collect { usr ->
+      username.value = usr ?: ""
+    }
+  }
+
+  fun saveUsername() {
+    viewModelScope.launch(dispatcher) {
+      usernameUseCase.saveUsername(username.value)
+    }
+  }
```
Comment (NickHuk) [inline settingsFeature/src/main/java/com/panhuk/settingsfeature/SettingsViewModel.kt]: create username repository and use it here, use case can be removed

File: usecaseImpl/src/main/java/com/panhuk/usecaseImpl/GetUsernameUseCaseImpl.kt [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import kotlinx.coroutines.flow.Flow
+
+class GetUsernameUseCaseImpl(private val dataStorePreferences: DatastorePreferences) :
+  GetUsernameUseCase {
+  override suspend fun saveUsername(username: String) {
+    dataStorePreferences.saveUsername(username)
+  }
+
+  override fun getUsername(): Flow<String?> {
+    return dataStorePreferences.observeUsername()
```
Comment (NickHuk) [inline usecaseImpl/src/main/java/com/panhuk/usecaseImpl/GetUsernameUseCaseImpl.kt]: Don't use data store preferences directly in use case, use common structure UserRepository -> UserDateSource -> DataStorePreferences

**Review by NickHuk — CHANGES_REQUESTED** [review]

File: menuFeature/src/main/java/com/panhuk/menufeature/MenuFragment.kt (hunk 1 above)
Comment (PanVova) [reply menuFeature/src/main/java/com/panhuk/menufeature/MenuFragment.kt]: +

File: core/src/main/java/com.panhuk.core/Fragments.kt [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package com.panhuk.core
+
+enum class Fragments {
```
Comment (NickHuk) [inline core/src/main/java/com.panhuk.core/Fragments.kt]: Will be better to rename this for ACTION or SCREEN. More abstract name not related to android Framework

**Review by NickHuk — CHANGES_REQUESTED** [review]
just one lil comment. fix it and feel free to merge

**Pull request merged**